spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/FailoverClientConnectionFactoryTests.java

```diff
@@ -193,10 +200,44 @@ public void testFailoverAllDead() throws Exception {
 		FailoverClientConnectionFactory failoverFactory = new FailoverClientConnectionFactory(factories);
 		failoverFactory.start();
 		GenericMessage<String> message = new GenericMessage<String>("foo");
-		failoverFactory.getConnection().send(message);
+		assertThatExceptionOfType(UncheckedIOException.class).isThrownBy(() ->
+			failoverFactory.getConnection().send(message));
 		Mockito.verify(conn2).send(message);
 	}
 
+	@Test
+	void failoverAllDeadAfterSuccess() throws Exception {
+		ServerSocket ss1 = ServerSocketFactory.getDefault().createServerSocket(0);
+		ServerSocket ss2 = ServerSocketFactory.getDefault().createServerSocket(0);
```

Comment:
Do we really need to do the **real** connection for this kind of test?
Won't just mocking target CCFs be enough?

This new test has failed on GH Actions workflow. But there was an OOM, so I'm not sure if that is a real failure...

Reply:
True - can mock the other factory - will do after I resolve the test race.